server/src/main/java/io/crate/planner/operators/JoinPlanBuilder.java

```diff
@@ -113,38 +112,26 @@ static LogicalPlan buildJoinTree(List<AnalyzedRelation> from,
             joinCondition = joinLhsRhs.condition();
         }
 
+        var joinSplitter = SymbolSplitter.of(joinCondition);
+        var validJoinConditions = AndOperator.join(joinSplitter.remainder(), null);
```

Comment:
I think you could just go with default value, as the default value would be true.
```
var validJoinConditions = AndOperator.join(joinSplitter.remainder());
```

Reply:
Then we don't need to extend join to a default value.

Reply:
This will cause `└ NestedLoopJoin[CROSS | true]` where the `true` is redundant. Should I try to remove that `true` differently?

Reply:
I think this is fine, just adapt the plan.